### PR TITLE
net-libs/libsignal-protocol-c: Fix CVE-2022-48468

### PR DESCRIPTION
**Pull request by bowlofeggs**
This commit fixes CVE-2022-48468 for this package's bundled protobuf-c.

Here are some reference links about the issue:

https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-48468 https://bugzilla.redhat.com/show_bug.cgi?id=2186673

For reference, here is the commit I made in Fedora to address the issue, which includes this patch:

https://src.fedoraproject.org/rpms/libsignal-protocol-c/c/152eb06d164e7973fda49139bc5a51f3b23c0cf6?branch=rawhide

Closes: https://bugs.gentoo.org/905098

### CODE REVIEW

**Comment by thesamesam** [comment]
Thanks!

File: net-libs/libsignal-protocol-c/libsignal-protocol-c-9999.ebuild [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright 1999-2020 Gentoo Authors
+# Copyright 1999-2023 Gentoo Authors
```
Comment (thesamesam) [inline net-libs/libsignal-protocol-c/libsignal-protocol-c-9999.ebuild:1]: Is it pending upstream or is upstream dead? If it's dead, we should drop 9999. If it's pending upstream, we should put a link in the patch.

Comment (bowlofeggs) [reply net-libs/libsignal-protocol-c/libsignal-protocol-c-9999.ebuild:1]: Hi @thesamesam,

Sadly the upstream for this package is archived, so there's no where to send the patch to: https://github.com/signalapp/libsignal-protocol-c

I'll send a commit soon to drop the -9999 patch.

Comment (bowlofeggs) [reply net-libs/libsignal-protocol-c/libsignal-protocol-c-9999.ebuild]: @thesamesam I have pushed up a new commit just before the CVE fix commit that drops the -9999 ebuild.

File: net-libs/libsignal-protocol-c/libsignal-protocol-c-2.3.3-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake
+
+DESCRIPTION="Signal Protocol C Library"
+HOMEPAGE="https://www.whispersystems.org/"
```
Comment (juippis) [inline net-libs/libsignal-protocol-c/libsignal-protocol-c-2.3.3-r1.ebuild:9]: This redirects.

File: net-libs/libsignal-protocol-c/libsignal-protocol-c-2.3.3-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
```
Comment (juippis) [inline net-libs/libsignal-protocol-c/libsignal-protocol-c-2.3.3-r1.ebuild:4]: 8!

**Review by juippis — APPROVED** [review]
I'll update the notes, thanks!

**Pull request closed without merging**